### PR TITLE
Cleanup: use Dune 2.0; reformat

### PR DESCRIPTION
**Pull request by lindig**
This is just some cleanup to access modern Dune features. If you prefer a different code formatting I can adjust to that, of course. I tried to pick something that is not controversial.

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -1,2 +1,34 @@
...
+(maintainers "Hannes Mehnert <hannes@mehnert.org>")
+
+(package
+ (name xenstore)
+ (synopsis "Xenstore protocol in pure OCam")
+ (description
+  "Implementation of the xenstore protocol in OCaml. This is an\n  alternative implementation to the implementation in the Xen project.")
+ (depends
+  (dune
+   (>= "2.0"))
```
Comment (edwintorok) [inline dune-project:28]: Dune recommends to leave a dependency on dune out: it'll automatically add a dependency based on the 'lang' version declared in dune-project.

Comment (edwintorok) [reply dune-project:28]: See https://dune.readthedocs.io/en/stable/howto/opam-file-generation.html#general-notes-and-tips

File: xenstore.opam [hunk 2]
```diff
@@ -1,35 +1,38 @@
+# This file is generated by dune, edit dune-project instead
 opam-version: "2.0"
-maintainer: "dave@recoil.org"
+synopsis: "Xenstore protocol in pure OCam"
+description: """
```
Comment (edwintorok) [inline xenstore.opam:4]: There used to be a longer description here, see the previous .opam file, can we keep it?

Comment (lindig) [reply xenstore.opam:4]: Ahh - missed it. Will fix

**Review by edwintorok — APPROVED** [review]
Thanks, looks good in general. I don't have a particular opinion on what `ocamlformat` settings should be used, what matters to me is that formatting is consistent within a project, and using `ocamlformat` ensures that.

If @freddy77 is working on more patches to ocaml-xenstore, then probably a good idea to rebase that work once this PR is merged to avoid conflicts due to formatting
(although not a big problem if a PR is created on the old code, we should be able to reformat the PR using the `merge-fmt` tool).

**Comment by freddy77** [comment]
> Thanks, looks good in general. I don't have a particular opinion on what `ocamlformat` settings should be used, what matters to me is that formatting is consistent within a project, and using `ocamlformat` ensures that.
> 
> If @freddy77 is working on more patches to ocaml-xenstore, then probably a good idea to rebase that work once this PR is merged to avoid conflicts due to formatting (although not a big problem if a PR is created on the old code, we should be able to reformat the PR using the `merge-fmt` tool).

No problem merging this in, maximum I'll reformat my code too and rebase on it.

File: Makefile [hunk 3]
```diff
@@ -15,3 +15,10 @@ uninstall:
 
 clean:
 	dune clean
+
+format:
+	dune build --auto-promote @fmt
+	dune format-dune-file dune-project > $$$$ && mv $$$$ dune-project
+	opam lint
+
+
```
Comment (freddy77) [inline Makefile:24]: Minor: why these empty lines?

**Comment by freddy77** [comment]
@lindig @edwintorok why this was not merged? Can we merge it ?

**Comment by edwintorok** [comment]
I don't have permission to merge PRs on this repo, I'm not sure who does.

**Comment by edwintorok** [comment]
It looks like your previous PR got meged by @palainp, perhaps he can help merge this PR too?

**Comment by palainp** [comment]
Sorry all, I missed that one, I merge now :)

**Comment by palainp** [comment]
Thank you!

And BTW, looking at https://github.com/mirage/ocaml-xenstore/pull/53#issuecomment-2208634921,  feel free to ping me when there are enough changes to cut a new release :)

**Pull request merged**